x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -666,6 +668,33 @@ static DeprecationIssue checkClusterRoutingAllocationIncludeRelocationsSetting(f
         );
     }
 
+    static DeprecationIssue checkFractionalByteValueSettings(final Settings settings,
+                                                             final PluginsAndModules pluginsAndModules,
+                                                             final ClusterState clusterState,
+                                                             final XPackLicenseState licenseState) {
+        Map<String, String> fractionalByteSettings = new HashMap<>();
+        for (String key : settings.keySet()) {
+            try {
+                settings.getAsBytesSize(key, ByteSizeValue.ZERO);
+                String stringValue = settings.get(key);
+                if (stringValue.contains(".")) {
+                    fractionalByteSettings.put(key, stringValue);
+                }
+            } catch (Exception ignoreThis) {
```

Comment:
Hmm, I'm not a huge fan of Exception based control flow. I know that settings are registered with ClusterSettings and IndexScopedSettings so that settings can be validated against the master set, but I don't know if we can filter on the setting's data type due to type erasure, and I'm not sure if those are comprehensive for every setting either.